ProcessLib/LIE/HydroMechanics/HydroMechanicsProcess.cpp

```diff
@@ -489,9 +490,22 @@ void HydroMechanicsProcess<GlobalDim>::computeSecondaryVariableConcrete(
 
     // compute nodal w and aperture
     auto const& R = _process_data.fracture_property->R;
+    auto const& b0 = _process_data.fracture_property->aperture0;
     MeshLib::PropertyVector<double>& vec_w = *_process_data.mesh_prop_nodal_w;
     MeshLib::PropertyVector<double>& vec_b = *_process_data.mesh_prop_nodal_b;
 
+    auto compute_nodal_aperture = [&](std::size_t const node_id,
+                                      double const w_n) {
+        // skip aperture computation for element-wise defined b0 because there
+        // are jumps on the nodes between the element's values.
+        if (dynamic_cast<MeshElementParameter<double> const*>(b0))
```

Comment:
Why not fail immediately, because the NaN will destroy the computation anyway?
Btw: Can't that be checked during the initialization?

Reply:
The NaN is for output only, the computation is not using this value.
One could check it in the initialization, but then it's another boolean indicating some special feature.
I'd like to keep this output-only thing here.

Reply:
OK.